md/moves.py

```diff
@@ -56,3 +68,197 @@ def n_accepted(self):
     @property
     def n_proposed(self):
         return np.sum(self.n_proposed_by_move)
+
+
+class NPTMove(MonteCarloMove):
+    def __init__(
+        self,
+        ubps,
+        lamb,
+        masses,
+        temperature,
+        pressure,
+        n_steps,
+        dt=1.5e-3,
+        friction=1.0,
+        barostat_interval=5,
+        seed=None,
+    ):
+
+        intg = lib.LangevinIntegrator(temperature, dt, friction, masses, seed)
+        self.integrator_impl = intg.impl()
+        all_impls = [bp.bound_impl(np.float32) for bp in ubps]
+
+        bond_list = get_bond_list(ubps[0])
+        group_idxs = get_group_indices(bond_list)
+
+        barostat = lib.MonteCarloBarostat(len(masses), pressure, temperature, group_idxs, barostat_interval, seed + 1)
+        barostat_impl = barostat.impl(all_impls)
+
+        self.bound_impls = all_impls
+        self.barostat_impl = barostat_impl
+        self.lamb = lamb
+        self.n_steps = n_steps
+
+    def move(self, x: CoordsVelBox):
+        # note: context creation overhead here is actually very small!
+        ctxt = custom_ops.Context(
+            x.coords, x.velocities, x.box, self.integrator_impl, self.bound_impls, self.barostat_impl
+        )
```

Comment:
Nice! Yeah, I think we had measured this to be ~0.2 ms  -- negligible when `n_steps` is 100s of MD steps. May want to print a warning in the constructor if `n_steps` is very small (say, < 10?)